### PR TITLE
Jetpack speed up settings

### LINKED ISSUE #7196: Jetpack Settings: Include lazy-load and photon modules active/inactive setting

### PR DESCRIPTION
**Pull request by planarvoid**
Fixes #7196 
Add "Speed up" settings to Site settings. There are two options with descriptions. One is to load images lazily and the second to server images from our servers. Both have been added to the "General" tab. Settings are only visible on Jetpack version >= 5.8

To test:

* Log in with site connected to Jetpack with version >= 5.8
* Go to site settings
* Click on "Speed up your site" 
* See 2 new settings with description
  * Both are clickable and are saved when clicked

screenshot_1519051194

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/DotComSiteSettings.java [hunk 1]
```diff
@@ -132,6 +144,28 @@ protected void fetchRemoteData() {
         }
     }
 
+    static boolean supportsJetpackSpeedUpSettings(SiteModel site) {
+        String jetpackVersion = site.getJetpackVersion();
```
Comment (hypest) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/DotComSiteSettings.java]: Seems like there is an opportunity here to generalize the "check minimum Jetpack version" code now that there is this second place we need to have it (first is in `PluginUtils.isPluginFeatureAvailable()`). What do you think about introducing a helper method in `SiteUtils` to do this?

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/DotComSiteSettings.java [hunk 2]
```diff
@@ -277,6 +316,44 @@ public void onErrorResponse(VolleyError error) {
...
+    private void fetchJetpackModuleSettings() {
+        ++mFetchRequestCount;
+        WordPress.getRestClientUtilsV1_1().getJetpackModuleSettings(
+                mSite.getSiteId(), new RestRequest.Listener() {
+                    @Override
+                    public void onResponse(JSONObject response) {
+                        AppLog.v(AppLog.T.API, "Received Jetpack module settings");
+                        if (response == null) return;
+                        JSONArray array = response.optJSONArray("modules");
+                        if (array == null) return;
```
Comment (hypest) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/DotComSiteSettings.java]: Not sure, should we try to emit `onFetchResponseReceived` before those `return` statements?

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/DotComSiteSettings.java (hunk 1 above)
Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/ui/prefs/DotComSiteSettings.java]: sounds great! I'll do that

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/DotComSiteSettings.java (hunk 2 above)
Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/ui/prefs/DotComSiteSettings.java]: makes sense! I checked the other calls and they seem to do that. I'll change the logic to always call the `onFetchResponseReceived`

**Comment by planarvoid** [comment]
@hypest thanks for the review! I've implemented both changes. 
Just a small detail: I have kept the `supportsJetpackSpeedUpSettings` static method even though it's pretty simple now. I feel like it adds some clarity (instead of inlining it).

File: WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java [hunk 3]
```diff
@@ -57,4 +58,32 @@ public static String getSiteIconUrl(SiteModel site, int size) {
...
+        if (site.isUsingWpComRestApi() && site.isJetpackConnected() && !TextUtils.isEmpty(jetpackVersion)) {
+            try {
+                // strip any trailing "-beta" or "-alpha" from the version
+                int index = jetpackVersion.lastIndexOf("-");
+                if (index > 0) {
+                    jetpackVersion = jetpackVersion.substring(0, index);
+                }
+                Version siteJetpackVersion = new Version(jetpackVersion);
+                Version minVersion = new Version(limitVersion);
+                return siteJetpackVersion.compareTo(minVersion) >= 0; // if the site has Jetpack 5.8 or newer installed
```
Comment (hypest) [inline WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java]: Nitpick: Let's update the comment here, just to avoid confusion.

Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java]: I removed the comment so it didn't really make sense in a generic way (the code itself is really self-explanatory)

**Comment by planarvoid** [comment]
@hypest all done!

**Comment by hypest** [comment]
🚢 it!

**Pull request merged**